app/models/patient_session.rb

```diff
@@ -49,4 +49,27 @@ def able_to_vaccinate?
     !unable_to_vaccinate? && !unable_to_vaccinate_not_assessed? &&
       !unable_to_vaccinate_not_gillick_competent?
   end
+
+  def consent_given?
+    return false if consent_no_response?
+
+    patient.consents.all?(&:response_given?)
+  end
```

Comment:
Doesn't this and `consent_refused?` override the methods defined in `PatientSessionStateMachineConcern`?

Reply:
You're right!